60-hippopotamus/technical_specification.md

```diff
@@ -0,0 +1,61 @@
+- Create the base Angular project files and directory structure using `ng new` with Angular v19.
+- Configure the application to run zoneless and without SSR according to [ADR017](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr017_server-side_rendering_in_angular.md).
+- Configure the project to use pnpm as a package manager as specified in [ADR015](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr015_node_runtime_selection.md).
+- Add pre-commit hooks and configuration for Prettier, ESLint and other useful linters.
+- Add `@angular-eslint/eslint-plugin` with the recommended rules as specified in [ADR013](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr013_angular_code_style.md).
+- Add subdirectories as specified in [ADR018](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr018_frontend_architecture.md).
+- Add `eslint-plugin-boundaries` configuration as specified in [ADR018](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr018_frontend_architecture.md).
+- Add and configure ESLint plugin to enforce documentation via JSDoc as specified in [ADR014](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr014_angular_project_documentation.md).
+- Add support for Compodoc as specified in [ADR014](https://github.com/ghga-de/adrs/blob/main/docs/adrs/).
+- Add the latest Angular Material version as dependency as specified in [ADR019](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr019_responsive_design_systems.md) and [ADR020](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr020_angular_component_library.md).
+- Add Tailwind CSS as a dependency as specified in [ADR022](adr022_db_migrations.md).
+- Add a Node.js run script that injects the runtime configuration and a configuration service that can be used by components and other services to fetch the configuration.
+- Create the base components (app component, header, footer, landing page) and style them according to our existing corporate design, make everything look similar to the legacy application.
+- Make sure the application uses the proper semantic tags according to [ADR016](https://github.com/ghga-de/adrs/blob/main/docs/adrs/adr016_semantic_web_technologies.md).
+- Add a minimal multi-stage production docker file that uses the run script to inject the configuration without rebuilding the application and SWS (`static-web-server`) to serve it as a single-page application.
+- Add and configure `cashew` for caching HTTP requests as a dependency.
+- Add and configure `MSW` (Mock Service Worker) for mocking the backend as a dependency.
+- Add authentication (login via LS Login, registration, 2FA) using `oidc-client-ts` (will be moved to the backend later). The profile page can be incomplete.
+- Components that require authentication should be guarded and lazy-loaded.
+- Make sure it is easy to switch between testing the application manually on localhost against MSW and against the staging or testing backend deployment.
```

Comment:
This is a bit vague. I think the most important part of this is to have it documented in a way that makes it easy to follow.

Reply:
True, improved in b100db27.